plugins/search-backend/src/service/router.ts

```diff
@@ -19,16 +19,42 @@ import Router from 'express-promise-router';
 import { Logger } from 'winston';
 import { SearchQuery, SearchResultSet } from '@backstage/search-common';
 import { SearchEngine } from '@backstage/plugin-search-backend-node';
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
 
 export type RouterOptions = {
   engine: SearchEngine;
   logger: Logger;
+  discovery: PluginEndpointDiscovery;
+  allowedLocationProtocols?: string[];
 };
 
+const defaultAllowedLocationProtocols = ['http:', 'https:'];
+
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { engine, logger } = options;
+  const {
+    engine,
+    logger,
+    discovery,
+    allowedLocationProtocols = defaultAllowedLocationProtocols,
+  } = options;
+  const baseUrl = await discovery.getExternalBaseUrl('');
```

Comment:
This isn't really a valid call and might break for some implementations. Perhaps something we should protect against in the current implementation too to set expectations. Not quite sure why this is needed though? We're just out to check what the protocol of the document location is right? Could just use example.com as the base

Reply:
I think I just took a notion initially that I could determine what the final URL might actually be and validate that but I agree it was a bad idea 😄. Just over thought it.